src/drivers/char/char_dev.c

```diff
@@ -46,18 +92,18 @@ int char_dev_register(const char *name, const struct kfile_operations *ops) {
 		return -ENODEV;
 	}
 
-	vfs_create_child(&node, name, S_IFCHR | S_IRALL | S_IWALL, &node);
+	vfs_create_child(&node, cdev->name, S_IFCHR | S_IRALL | S_IWALL, &node);
 	if (!(node.node)) {
 		return -1;
 	}
 
 	dev_nas = node.node->nas;
-	dev_nas->fs = filesystem_create("empty");
+	dev_nas->fs = filesystem_create("devfs");
 	if (dev_nas->fs == NULL) {
 		return -ENOMEM;
 	}
-
-	dev_nas->fs->file_op = ops;
+	
+	node.node->nas->fi->privdata = (void *) cdev;
```

Comment:
No need to cast `cdev` to `void *` (for `void *` pointer of any type is ok, no warnings here).

Reply:
If it's used 'const' than it's required